core/src/main/java/org/bitcoinj/crypto/OutputDescriptor.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.crypto;
```

Comment:
Somehow it doesn't feel right this class in the crypto package. I'd expect it either in wallet or in base, or maybe even in its own "descriptor" module if we implemented everything an OD can do.

Reply:
> Somehow it doesn't feel right this class in the crypto package.

It is a serialization format for public and private keys and the current implementation requires classes in the crypto package/module. 

There are several options for moving forward:

1. Move the outer interface class to the `base` _module_ and make the inner implementation class a standalone class in either the `crypto` or `core` (or for 0.19, `wallet`) _module_.

2. Make a `descriptors` _package_ in any of our _modules_.

3. Make the `Output Descriptor` just a dumb data holder and put the code that requires classes from `crypto` somewhere else entirely. The argument against this is that it is convenient to parse/validate the descriptor into our native representation of its components.

And I believe these 3 options are largely independent of each other.